res/controllers/Pioneer-DDJ-400-script.js

```diff
@@ -265,10 +265,14 @@ PioneerDDJ400.beatFxRightPressed = function(_channel, _control, value) {
 
 PioneerDDJ400.beatFxSelectPressed = function(_channel, _control, value) {
     if (value === 0) { return; }
-
-    engine.setValue("[EffectRack1_EffectUnit1]", "focused_effect", 3);
+    var focusfx = engine.getValue("[EffectRack1_EffectUnit1]", "focused_effect")+1;
+    if (engine.getValue("[EffectRack1_EffectUnit1]", "focused_effect")===3){
```

Comment:
Ok, I see, the code is confusing, there are 3 effects.

- engine.getValue() should only be called once and assigned to variable, e.g. focused_effect
- if focused_effect >== 3 { focused_effect = 1 } else { focused_effect = focused_effect + 1 }